core/src/components/refresher/test/a11y/refresher.e2e.ts

```diff
@@ -0,0 +1,32 @@
+import AxeBuilder from '@axe-core/playwright';
+import { expect } from '@playwright/test';
+import { configs, test, dragElementBy } from '@utils/test/playwright';
+
+configs({ directions: ['ltr'], modes: ['md'], themes: ['light', 'dark'] }).forEach(({ config, title }) => {
+  test.describe(title('refresher: a11y'), () => {
+    test('should not have accessibility violations', async ({ page }) => {
+      await page.setContent(
+        `
+        <ion-content>
+          <ion-refresher class="refresher-native" slot="fixed">
+            <ion-refresher-content></ion-refresher-content>
+          </ion-refresher>
+        </ion-content>
+      `,
+        config
+      );
+
+      const target = page.locator('body');
+
+      await page.waitForSelector('ion-refresher.hydrated', { state: 'attached' });
+
+      await dragElementBy(target, page, 0, 320, undefined, undefined, false);
+
+      await page.waitForSelector('.arrow-container', { state: 'attached' });
+
+      const results = await new AxeBuilder({ page }).disableRules('region').analyze();
```

Comment:
The Axe test fails because a landmark role like `<main> is missing. However, when I add it in then it fails for having a nested `<main>`. The `ion-content` adds it in but still fails because it's not at the top level.

Reply:
I'm not sure I understand this comment. `setContent` adds the HTML as a direct descendant of `body`, so there shouldn't be a nested `main` element: https://github.com/ionic-team/ionic-framework/blob/e886e3ff2fcb8a3586a62881c5fc848f3074235d/core/src/utils/test/playwright/page/utils/set-content.ts#L57

Reply:
The nested (second) `<main>` comes from `ion-content` [logic](https://github.com/ionic-team/ionic-framework/blob/c680ad99885ce7600d7447a5aadedd2b3f79adf9/core/src/components/content/content.tsx#L431).

So the following code:
```ts
await page.setContent(
  `
  <main>
      <ion-content>
      <ion-refresher class="refresher-native" slot="fixed">
        <ion-refresher-content></ion-refresher-content>
      </ion-refresher>
    </ion-content>
   </main>
`,
  config
);
```

would be:
```tsx
<main> // This main solves the violation for not having a landmark
   <ion-content>
       <main> // This main triggers a violation for being nested
           // stuff
       </main>
    </ion-content>
</main>
```

Reply:
Just so I understand: Only using `ion-content` is not sufficient because `main` is nested within `ion-content` when it really should be the other way around.

What about adding `role="main"` to the host of `ion-content`? That might be a better solution that nested a `main` element inside of the Shadow DOM.

Reply:
I don't think we need to make any changes in this PR, but we should make a tech debt ticket to evaluate doing `<ion-content role="main">` from `content.tsx` instead of nesting the `<main>` element. We can then add a `TODO` comment above this disabled rule.